src/dotnet/commands/dotnet-build/CompileContext.cs

```diff
@@ -328,10 +328,10 @@ private bool InvokeCompileOnRootProject()
                 args.Add(_args.ArchValue);
             }
 
-            if (!string.IsNullOrWhiteSpace(_args.IlcArgsValue))
+            foreach (var ilcArg in _args.IlcArgsValue)
             {
-                args.Add("--ilcargs");
-                args.Add(_args.IlcArgsValue);
+                args.Add("--ilcarg");
```

Comment:
We write response files now and in the file, you have to have arguments in this format:

```
-option1:value1
-option2:value2
```

With a opaque string like `--ilcargs`, we would end up with in the response file.
```
-r:c:\foo.dll -r:c:\bar.dll -r:c:\baz.dll
```

unless you write a parser that resolves these into per ilc arg in the above format and separates them with a newline.

With `--ilcarg` when you chain each arg, you end up with:

```
-r:c:\foo.dll
-r:c:\bar.dll
-r:c:\baz.dll
```

in the response file. @terrajobst any other suggestions, we pass in `--ilcargs` as an opaque string through a response file to ILC from `dotnet compile-native`